src/main/scala/sbtavro/SbtAvro.scala

```diff
@@ -62,9 +90,51 @@ object SbtAvro extends AutoPlugin {
     avroUseNamespace := false
   )
 
-  override lazy val projectSettings: Seq[Setting[_]] =
+  override lazy val projectSettings: Seq[Setting[_]] = defaultSettings ++
     Seq(Compile, Test).flatMap(c => inConfig(c)(configScopedSettings))
 
+  private def unpack(deps: Seq[(ModuleID, File)],
+                     extractTarget: File,
+                     streams: TaskStreams): Seq[File] = {
+    def cachedExtractDep(module: ModuleID, jar: File): Seq[File] = {
+      val cached = FileFunction.cached(
+        streams.cacheDirectory / jar.name,
+        inStyle = FilesInfo.lastModified,
+        outStyle = FilesInfo.exists
+      ) { deps =>
+        val moduleTarget = extractTarget / module.organization / module.name
```

Comment:
This is a trade-off: Unpack on separate directories to avoid file name conflict between deps and fail avro generation in case `avroUseNamespace` is set. Or unpack all schemas in `extractTarget` but risk conflict